TileDB test failure with tiledb 2.27 on Windows Conda

@sgillies Any desire to take care of TileDB related stuff ;-) ?

master test failure in https://github.com/OSGeo/gdal/actions/runs/12298734799/job/34322855248:
```
2024-12-12T15:09:08.2457022Z 38: FAILED tiledb_write.py::test_tiledb_write_overviews - RuntimeError: Array::delete_array(C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-2\test_tiledb_write_overviews0\test.tiledb\l_1) failed: [TileDB::IO] Error: Failed to delete path 'C:\Users\runneradmin\AppData\Local\Temp\pytest-of-runneradmin\pytest-2\test_tiledb_write_overviews0\test.tiledb\l_1'; not a valid path.
```

Recent successful builds were on tiledb 2.26.2  (https://github.com/OSGeo/gdal/actions/runs/12287644512/job/34290038383), so this is 2.27.0 specific

--- a/autotest/gdrivers/tiledb_write.py
+++ b/autotest/gdrivers/tiledb_write.py
@@ -451,10 +451,6 @@ def test_tiledb_write_create_group(tmp_path):
     assert ds.RasterYSize == 2
 
 
-@pytest.mark.skipif(
-    gdaltest.is_travis_branch("build-windows-conda"),
-    reason="Fails with tiledb 2.27. Seehttps://github.com/OSGeo/gdal/issues/11485",
-)
 def test_tiledb_write_overviews(tmp_path):
 
     # This dataset name must be kept short, otherwise strange I/O errors will

--- a/frmts/tiledb/tiledbdense.cpp
+++ b/frmts/tiledb/tiledbdense.cpp
@@ -2933,7 +2933,9 @@ CPLErr TileDBRasterDataset::IBuildOverviews(
             {
                 CPL_IGNORE_RET_VAL(poODS->Close());
                 tiledb::Array::delete_array(*m_ctx, poODS->GetDescription());
-                vfs.remove_dir(poODS->GetDescription());
+                if (vfs.is_dir(poODS->GetDescription()) {
+                    vfs.remove_dir(poODS->GetDescription());
+                }
             }
             catch (const tiledb::TileDBError &e)
             {

--- a/frmts/tiledb/tiledbdense.cpp
+++ b/frmts/tiledb/tiledbdense.cpp
@@ -2933,7 +2933,8 @@ CPLErr TileDBRasterDataset::IBuildOverviews(
             {
                 CPL_IGNORE_RET_VAL(poODS->Close());
                 tiledb::Array::delete_array(*m_ctx, poODS->GetDescription());
-                if (vfs.is_dir(poODS->GetDescription())) {
+                if (vfs.is_dir(poODS->GetDescription()))
+                {
                     vfs.remove_dir(poODS->GetDescription());
                 }
             }

--- a/frmts/tiledb/tiledbdense.cpp
+++ b/frmts/tiledb/tiledbdense.cpp
@@ -2933,7 +2933,7 @@ CPLErr TileDBRasterDataset::IBuildOverviews(
             {
                 CPL_IGNORE_RET_VAL(poODS->Close());
                 tiledb::Array::delete_array(*m_ctx, poODS->GetDescription());
-                if (vfs.is_dir(poODS->GetDescription()) {
+                if (vfs.is_dir(poODS->GetDescription())) {
                     vfs.remove_dir(poODS->GetDescription());
                 }
             }

